include/ginkgo/core/base/executor.hpp

```diff
@@ -104,12 +104,42 @@ constexpr allocation_mode default_hip_alloc_mode =
 inline namespace cl {
 namespace sycl {
 
+
 class queue;
 
+
 }  // namespace sycl
 }  // namespace cl
 
 
+/**
+ * The enum class for the dpcpp queue property.
+ */
+enum class dpcpp_queue_property {
+    /**
+     * queue executes in order
+     */
+    in_order = 1,
+
+    /**
+     * queue enables the profiling
+     */
+    enable_profiling = 2,
+
+    /**
+     * queue enables the profiling and executes in order
+     */
+    in_order_and_enabling_profiling = 3
```

Comment:
This can be removed (see my previous reply: https://github.com/ginkgo-project/ginkgo/pull/1110#discussion_r989077382) since it is legal for an `enum class` to have values other than specified in the enumerator-list (as long as it fits in the underlying type).